risc0/zkvm/src/host/server/exec/syscall/fork.rs

```diff
@@ -111,10 +111,13 @@ impl<'a, 'b> ChildExecutor<'a, 'b> {
     fn ecall_software(&mut self) -> Result<bool> {
         tracing::debug!("ecall_software");
         let into_guest_ptr = ByteAddr(EmuContext::load_register(self, REG_A0)?);
-        if !is_guest_memory(into_guest_ptr.0) && !into_guest_ptr.is_null() {
+        let into_guest_len = EmuContext::load_register(self, REG_A1)?;
+        if into_guest_len > 0
+            && !(is_guest_memory(into_guest_ptr.0)
+                && is_guest_memory(into_guest_ptr.0 + into_guest_len))
+        {
```

Comment:
Note that this _is_ a slightly different check now. I noticed this is different than the main emulator, but it's unclear why it would be. @flaub , is there a reason the check here should be different than the main emulator's `ecall_software`?

Reply:
Probably just forgot to update fork. The main thing is that the rust compiler is allowed to supply an unaligned pointer when the length is 0, so the main emulator was updated to account for this.

Reply:
Ditto here about checked_add

Reply:
I've removed these changes.